extensions/controllers/sandboxclaim_controller_test.go

```diff
@@ -363,6 +374,343 @@ func TestSandboxClaimReconcile(t *testing.T) {
+						Status: status,
+						Reason: reason,
+					},
+				},
+			},
+		}
+	}
+
+	testCases := []struct {
+		name               string
+		claim              *extensionsv1alpha1.SandboxClaim
+		sandboxIsExpired   bool // If false, Sandbox is still Running
+		expectClaimDeleted bool
+		expectStatus       string
+	}{
+		{
+			name:               "Policy=Retain -> Should Retain Claim even if time passed",
+			claim:              createClaim("retain-claim", extensionsv1alpha1.ShutdownPolicyRetain),
+			sandboxIsExpired:   true,
+			expectClaimDeleted: false,
+			expectStatus:       extensionsv1alpha1.ClaimExpiredReason,
+		},
+		{
+			name:               "Policy=Delete && Sandbox Expired -> Should Delete Claim",
+			claim:              createClaim("delete-claim-synced", extensionsv1alpha1.ShutdownPolicyDelete),
+			sandboxIsExpired:   true,
+			expectClaimDeleted: true,
+			expectStatus:       "",
+		},
+		{
+			name:               "Policy=Delete && Sandbox Running -> Should Delete Claim immediately (Fix Race Condition)",
+			claim:              createClaim("delete-claim-race", extensionsv1alpha1.ShutdownPolicyDelete),
+			sandboxIsExpired:   false, // Sandbox is still running!
+			expectClaimDeleted: true,  // But Claim should delete itself anyway
+			expectStatus:       "",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			scheme := newScheme(t)
+			sandbox := createSandbox(tc.claim.Name, tc.sandboxIsExpired)
+			client := fake.NewClientBuilder().WithScheme(scheme).
+				WithObjects(template, tc.claim, sandbox).
+				WithStatusSubresource(tc.claim).Build()
+
+			reconciler := &SandboxClaimReconciler{
+				Client:   client,
+				Scheme:   scheme,
+				Recorder: record.NewFakeRecorder(10),
+			}
+
+			req := reconcile.Request{NamespacedName: types.NamespacedName{Name: tc.claim.Name, Namespace: "default"}}
+			_, err := reconciler.Reconcile(context.Background(), req)
+			if err != nil {
+				t.Fatalf("reconcile failed: %v", err)
+			}
+
+			var fetchedClaim extensionsv1alpha1.SandboxClaim
+			err = client.Get(context.Background(), req.NamespacedName, &fetchedClaim)
+
+			if tc.expectClaimDeleted {
+				if !k8errors.IsNotFound(err) {
+					t.Errorf("Expected Claim to be deleted, but it still exists")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Expected Claim to exist, but got error: %v", err)
+				}
+				// Verify Status Message for Retained Claims
+				foundReason := false
+				for _, cond := range fetchedClaim.Status.Conditions {
+					if cond.Type == string(sandboxv1alpha1.SandboxConditionReady) && cond.Reason == tc.expectStatus {
+						foundReason = true
+					}
+				}
+				if !foundReason {
+					t.Errorf("Expected status reason %q, but not found", tc.expectStatus)
+				}
+			}
+		})
+	}
+}
+
+// TestSandboxRevivalAndEvents verifies:
+// 1. Sandbox creation emits "SandboxProvisioned"
+// 2. Updating Claim Lifecycle triggers a Sandbox Update (Sync)
+func TestSandboxRevivalAndEvents(t *testing.T) {
```

Comment:
The term "revival" could be misleading. A sandbox/claim shouldn't be revived after it has been marked expired. However, the shutdownTime can be extended (best effort) before they're marked expired.